src/kaggle_benchmarks/assertions.py

```diff
@@ -397,3 +417,123 @@ def assert_raises_no_exceptions(
+        for item in self.results:
+            if isinstance(item, dict):
+                cleaned_results.append(AssessResult(**item))
+            elif isinstance(item, AssessResult):
+                cleaned_results.append(item)
+            else:
+                continue
+        self.results = cleaned_results
+
+
+def generate_schema_description(schema_cls: Type[Any]) -> str:
+    """Generates a pretty-printed JSON schema string from a class."""
+    adapter = TypeAdapter(schema_cls)
+    schema = adapter.json_schema()
+    return json.dumps(schema, indent=2)
+
+
+def assess_response_with_judge(
+    criteria: Iterable[str],
+    response_text: str,
+    judge_llm: Any,
+    prompt_fn: Callable[[Iterable[str], str], str] | None = None,
+    output_schema: Type[Any] = AssessReport,
+) -> Any:
+    """
+    Assess if an LLM response satisfies requirements using a Judge LLM.
+
+    Args:
+        criteria: A list of requirements or facts the response must satisfy.
+        response_text: The actual output generated by the LLM under test.
+        judge_llm: The LLM interface object. Must support .prompt(text, schema=...).
+        prompt_fn: Optional callable with signature `(criteria, response_text) -> str`
+                   to generate a custom prompt. Recommended when using a custom
+                   `output_schema` to ensure the judge is aware of the format.
+        output_schema: The Data Class (schema) the LLM should return.
+                       Defaults to AssessReport containing AssessResult.
+
+    Returns:
+        An instance of output_schema (e.g., AssessReport).
+    """
+
+    if prompt_fn is None:
+        formatted_criteria = "\n".join(f"{i + 1}. {c}" for i, c in enumerate(criteria))
+        schema_desc = generate_schema_description(output_schema)
+
+        prompt_text = textwrap.dedent(f"""
+        You are an impartial and strict technical evaluator.
+
+        ### Task
+        Verify if the **Generated Response** complies strictly with the **Criteria**.
+
+        ### Input Data
+        **Generated Response:**
+        {response_text}
+
+        **Criteria:**
+        {formatted_criteria}
+
+        ### Evaluation Protocol
+        1. Analyze the response against the criteria.
+        2. Be strict. If the text is ambiguous, the check fails.
+        3. Output your assessment using the specific structure requested below.
+
+        ### Output Schema Requirements
+        You must return a JSON object containing the following fields:
+        {schema_desc}
+
+        *Ensure your JSON output strictly adheres to these types.*
+        """)
+    else:
+        prompt_text = prompt_fn(criteria, response_text)
+
+    try:
+        with chats.new(f"Response assessment with {judge_llm.name}"):
+            assess_report = judge_llm.prompt(prompt_text, schema=output_schema)
+
+        if isinstance(assess_report, dict):
+            assess_report = output_schema(**assess_report)
+
+    except Exception:
```

Comment:
While this isn't common during scheduled execution, I often rely on the `KeyboardInterrupt` exception to cancel running cells in a notebook.
```suggestion
    except KeyboardInterrupt: 
         raise 
    except Exception:
```